app/views/categories/_form.html.erb

```diff
@@ -3,35 +3,55 @@
 <div data-controller="color-avatar">
   <%= styled_form_with model: category, class: "space-y-4" do |f| %>
     <section class="space-y-4">
-      <div class="w-fit m-auto">
-        <%= render partial: "shared/color_avatar", locals: { name: category.name, color: category.color } %>
-      </div>
+      <details>
+        <summary>
+          <div class="w-fit m-auto">
+            <%= render partial: "shared/color_avatar", locals: { name: category.name, color: category.color } %>
+          </div>
+        </summary>
 
-      <div class="flex gap-2 items-center justify-center" data-color-avatar-target="selection">
-        <% Category::COLORS.each do |color| %>
-          <label class="relative">
-            <%= f.radio_button :color, color, class: "sr-only peer", data: { action: "change->color-avatar#handleColorChange" } %>
-            <div class="w-6 h-6 rounded-full cursor-pointer peer-checked:ring-2 peer-checked:ring-offset-2 peer-checked:ring-blue-500" style="background-color: <%= color %>"></div>
-          </label>
-        <% end %>
-      </div>
+        <div class=" absolute z-50 bg-white p-4 border border-alpha-black-25 rounded-2xl shadow-xs h-fit left-62" data-controller="color-picker">
+          <div class="flex gap-2 flex-col mb-4" data-color-avatar-target="selection">
+            <div data-color-picker-target="pickerSection"></div>
+            <h4 class="text-gray-500 text-sm">Color</h4>
+            <div class="flex gap-2 items-center" data-color-picker-target="colorsSection">
+              <% Category::COLORS.each do |color| %>
+                <label class="relative">
+                  <%= f.radio_button :color, color, class: "sr-only peer", data: { action: "change->color-picker#handleColorChange change->color-avatar#handleColorChange" } %>
+                  <div class="w-6 h-6 rounded-full cursor-pointer peer-checked:ring-2 peer-checked:ring-offset-2 peer-checked:ring-blue-500" style="background-color: <%= color %>"></div>
+                </label>
+              <% end %>
+              <div class="w-6 h-6 rounded-full cursor-pointer" data-color-picker-target="pickerBtn" style="background: conic-gradient(red,orange,yellow,lime,green,teal,cyan,blue,indigo,purple,magenta,pink,red)" ></div>
+            </div>
+            <div class="flex gap-2 items-center hidden flex-col" data-color-picker-target="paletteSection">
+              <div class="flex gap-2 items-center w-full">
+                <div class="w-6 h-6 p-4 rounded-full cursor-pointer" style="background-color: <%= category.color %>" data-color-picker-target="colorPreview"></div>
+                <%= f.text_field :color , data: { color_picker_target: "colorInput", color_avatar_target: "colorInput", color_picker_color_value: "color" }, class: "form-field__input w-auto" %>
```

Comment:
There are a few style mismatches here.  In order to clear out the default `styled_form` input styles, you can pass `inline: true` which tells our form builder to omit default styles:

```erb
<%= f.text_field :color , data: { color_picker_target: "colorInput", color_avatar_target: "colorInput", color_picker_color_value: "color" }, inline: true %>
```